content/courses/native-onchain-development/paging-ordering-filtering-data-frontend.md

```diff
@@ -147,21 +180,32 @@ For example, you could search through a list of contacts by including a `memcmp`
 filter:
 
 ```tsx
+const DATA_OFFSET = 2; // The first 2 bytes of metadata they store versioning information and are not needed for the search.
+const DATA_LENGTH = 18; // Retrieve 18 bytes of relevant data - This length includes the specific part of the account's data that holds the necessary information for comparison.
 async function fetchMatchingContactAccounts(
   connection: web3.Connection,
   search: string,
 ): Promise<(web3.AccountInfo<Buffer> | null)[]> {
```

Comment:
Please change these import to use `import { Connection } from "@solana/web3.js";` etc - it saves people from typing `web3` all the time.

Reply:
Please use generic syntax like `Promise<Array<AccountInfo<Buffer> | null>>` instead of `Promise<(AccountInfo<Buffer> | null)[]>` - it's easier to read as the parent always contains the child.

This is old code we're fixing here, so thanks for your patience!

Reply:
> Please change these import to use `import { Connection } from "@solana/web3.js";` etc - it saves people from typing `web3` all the time.

fixed✅

Reply:
> Please use generic syntax like `Promise<Array<AccountInfo<Buffer> | null>>` instead of `Promise<(AccountInfo<Buffer> | null)[]>` - it's easier to read as the parent always contains the child.
> 
> This is old code we're fixing here, so thanks for your patience!

fixed ✅

Reply:
I can still see `Promise<(web3.AccountInfo<Buffer> | null)[]>` above. Use `Array<>` not `[]`.